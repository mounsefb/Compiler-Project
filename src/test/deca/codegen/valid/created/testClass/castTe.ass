	TSTO #11
	BOV stack_overflow_error
	ADDSP #24
; start main program
; Main program
; Beginning of main instructions:
	LOAD #42, R2
	STORE R2, 1(GB)
	LOAD #0x1.4p2, R2
	STORE R2, 2(GB)
	LOAD 1(GB), R2
	LOAD #0, R3
	CMP R3, R2
	BGT greater1
	LOAD #0, R2
	BRA notGreater1
greater1:
	LOAD #1, R2
notGreater1:
	STORE R2, 3(GB)
	LOAD 1(GB), R2
	STORE R2, 4(GB)
	LOAD 2(GB), R2
	STORE R2, 5(GB)
	LOAD 3(GB), R2
	STORE R2, 6(GB)
	LOAD 4(GB), R1
	WINT
	WNL
	LOAD 5(GB), R1
	WFLOAT
	WNL
	LOAD 6(GB), R2
	CMP #1, R2
	BNE else2
then2:
	WSTR "c works"
	WNL
	BRA end_If2
else2:
end_If2:
	HALT
; end main program
overflow_error:
	WSTR "Error: Overflow during arithmetic operation"
	WNL
	ERROR
stack_overflow_error:
	WSTR "Error: Stack Overflow "
	WNL
	ERROR
io_error:
	WSTR "Error: Input/Output error"
	WNL
	ERROR
Dereferencement.null:
	WSTR "Erreur : dereferencement de null"
	WNL
	ERROR
pile_pleine:
	WSTR "Erreur : pile pleine"
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : allocation impossible, tas plein"
	WNL
	ERROR
cast_incorrect:
	WSTR "la conversion (cast) est incorrecte"
	WNL
	ERROR
; code de l'initialisation des champs
; code des m√©thodes
code.object.equals:
	LOAD -2(LB), R0
	CMP -3(LB), R0
	BEQ if_equal_pour_object.equals
	LOAD #0, R0
	BRA fin_methodes
if_equal_pour_object.equals:
	LOAD #1, R0
	BRA fin_methodes
fin_methodes:
	RTS
