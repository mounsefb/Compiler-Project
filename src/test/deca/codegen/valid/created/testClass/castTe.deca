{
    int x = 42;
    float y= 5.0;
    boolean z=  x > 0 ;
    int a = (int)(x);
    float b = (float)(y);  
    boolean c = (boolean)(z);
    println(a);
    println(b);
    if (c){
        println("c works");
    }
}