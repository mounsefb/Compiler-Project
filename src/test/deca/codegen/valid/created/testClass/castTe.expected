42
5.00000e+00
c works